code_generation/optimizer.py

```diff
@@ -136,42 +148,49 @@ def Optimize(self):
                 log.error("Please check, if all needed producers are activated")
                 raise Exception
             wrongProducer, wrong_inputs = self.check_ordering()
-            producers_to_relocate = self.find_inputs(wrongProducer, wrong_inputs)
-            if len(producers_to_relocate) == 0:
-                self.optimized = True
-            else:
-                for producer_to_relocate in producers_to_relocate:
-                    counter += 1
-                    self.relocate_producer(
-                        producer_to_relocate,
-                        self.get_position(producer_to_relocate),
-                        self.get_position(wrongProducer),
-                    )
+            if wrongProducer is not None:
```

Comment:
I think due to this if statement 
https://github.com/KIT-CMS/CROWN/blob/847581d22455fcc82e60e06b8752760b5cfc5e91/code_generation/optimizer.py#L152-L155
this if statement will never be true, because this would only happen if `find_inputs()` finds the input in the "global" scope, but this should already be covered by `check_ordering()`. Am I wrong?

Reply:
i think you are correct !